File: fsharp-csharp-interop-demos/unit-testing-demo/Domain.Tests.CSharp/EbayTests.cs
using FluentAssertions;
using Xunit;

namespace Domain.Tests.CSharp
{
    public class EbayTests
    {
        [Fact]
        public void GetSearchResultFor_returns_correct_result() =>
            Ebay.GetSearchResultsFor("foo").Should().Be("FOO");
    }
}

File: fsharp-csharp-interop-demos/unit-testing-demo/Domain.Tests.FSharp/EbayTests.fs
module EbayTests

open System
open FsUnit.Xunit
open Xunit
open Domain
open NHamcrest

[<Fact>]
let ``GetSearchResultFor returns correct result (using FsUnit)`` () =
    // Calling a C# static method (from a static class) from F#:
    Ebay.GetSearchResultsFor("bar") |> should equal "BAR"


[<Fact>]
let ``GetSearchResultFor returns correct result (not using FsUnit)`` () =
    Assert.That(Ebay.GetSearchResultsFor("bar"), Is.EqualTo("BAR"))
